examples/bitcoin_crate/src/main.rs

```diff
@@ -11,236 +11,175 @@ use bitcoin::{
-                long_term_feerate: Some(10.0),
-                min_absolute_fee: 1000 * (i + 1) as u64,
-                base_weight,
-                change_weight: 34,
-                change_cost: 5 + i as u64,
-                avg_input_weight: 148,
-                avg_output_weight: 34,
-                min_change_value: 100,
-                excess_strategy,
-            }
-        })
-        .collect()
-}
-
-fn perform_select_coin(utxos: Vec<OutputGroup>, coin_select_options_vec: Vec<CoinSelectionOpt>) {
-    // Printing information about the UTXOs used for selection
-    println!("\nThe total number of UTXOs available: {:?}", utxos.len());
-    for (i, utxo) in utxos.iter().enumerate() {
-        println!("\nUTXO #:{}", i);
-        println!("Value:{} sats", utxo.value);
-        println!("Weight:{} bytes", utxo.weight);
-        println!("No. of Inputs: {}", utxo.input_count);
+    for utxo in utxos.iter() {
         println!(
-            "Creation Sequence: {:?}",
+            "{:<15} | {:<15} | {:<15} | {:<20}",
+            utxo.value,
+            utxo.weight,
+            utxo.input_count,
             utxo.creation_sequence.unwrap_or(0)
         );
     }
+    println!("{:-<71}", "");
+}
 
-    for coin_select_options in coin_select_options_vec.iter().take(5) {
-        println!(
-            "\nSelecting UTXOs to total: {:?} sats",
-            coin_select_options.target_value
-        );
-        match select_coin(&utxos, coin_select_options) {
-            Ok(selectionoutput) => {
-                println!(
-                    "Selected utxo index and waste metrics are: {:?}",
-                    selectionoutput
-                );
-            }
-            Err(e) => {
-                println!("Error performing coin selection: {:?}", e);
-            }
-        }
-    }
+fn perform_select_coin(
+    utxos: &[OutputGroup],
+    opt: &CoinSelectionOpt,
+) -> Result<SelectionOutput, SelectionError> {
+    log_utxos(utxos);
+    select_coin(utxos, opt)
 }
 
 fn main() {
-    // Read and parse inputs from JSON file
-    let inputs = match read_json_file("examples/bitcoin_crate/txdata/txinp.json")
-        .and_then(|filedata| json_to_txin(&filedata))
-    {
-        Ok(txin_vec) => txin_vec,
-        Err(e) => {
-            println!("Error reading or parsing inputs: {:?}", e);
-            return;
-        }
+    let target: u64 = 4_000_000;
+
+    // Pre-calculate transaction weight using a dummy change output
+    // Rationale:
+    // 1. Transaction weight calculation requires both inputs and outputs
+    // 2. We know the target output value (4M sats) but not the change amount yet
+    // 3. Using 0 as placeholder for change output is valid because:
+    //    - Output weight in Bitcoin depends on script size, not value
+    //    - Value field is fixed 8 bytes regardless of amount
+    //    - Only the script pubkey size affects weight
+    let outputs = vec![create_txout(target), create_txout(0)];
```

Comment:
Remove the `create_txout` function and create two variables,
target_txout and change_txout

Reply:
done